### PR TITLE
[WIP] Fix add/sub decomposition

### LINKED ISSUE #4659: RyuJIT x86: using temps during long decomposition to avoid bad codegen
JIT\Methodical\AsgOp\i8\i8_cs_d\i8_cs_d.cmd
JIT\Methodical\AsgOp\i8\i8_cs_do\i8_cs_do.cmd
JIT\Methodical\AsgOp\i8\i8_cs_r\i8_cs_r.cmd
JIT\Methodical\AsgOp\i8\i8_cs_ro\i8_cs_ro.cmd

test output:

Initial parameters: x is -10 and y is 4.
f11     x += x - y failed.
x: -8589934616,         expected: -24

Initial parameters: x is -10 and y is 4.
f18     x += x ^ y failed.
x: -4294967320,         expected: -24

Initial parameters: x is -10 and y is 4.
f19     x += x | y failed.
x: -4294967316,         expected: -20
Expected: 100
Actual: 1

### PR DESCRIPTION
**Pull request by mikedn**
Fixes #4659, #4593

### CODE REVIEW

**Comment by mikedn** [comment]
This is more or less good to go once #5698, #5047, #5699, #5994 and possibly #5956 are fixed.

File: src/jit/lsra.cpp [hunk 1]
```diff
@@ -4564,6 +4564,8 @@ void LinearScan::setFrameType()
         }
     }
 
+    frameType = FT_EBP_FRAME;
```
Comment (pgavlin) [inline src/jit/lsra.cpp]: I'd prefer if we filed added a comment re: this workaround and checked this bit of the change in on its own.

**Comment by pgavlin** [comment]
LGTM aside from the note re: the workaround for #5699.

**Comment by pgavlin** [comment]
@dotnet/jit-contrib PTAL as well.

**Comment by CarolEidt** [comment]
It looks like there are two new failures in the x86/RyuJIT test run. I suspect that they may be exposing other issues once this is enabled, but it would be good to understand.
Otherwise, this looks good to me.

File: src/jit/lsra.cpp (hunk 1 above)
Comment (BruceForstall) [inline src/jit/lsra.cpp]: I presume also this should go under
```
#if defined(_TARGET_X86_)
```

Comment (mikedn) [inline src/jit/lsra.cpp]: This was added just to get tests working so I didn't bother too much with it. I think it should be in a different PR but maybe just a separate commit is fine (as long as the above feedback is addressed). Thoughts?

**Comment by adiaaida** [comment]
LGTM too!

File: src/jit/lsra.cpp (hunk 1 above)
Comment (pgavlin) [inline src/jit/lsra.cpp]: Let's make this change in a separate PR. Thanks!

**Comment by mikedn** [comment]
Well, I removed that workaround and it looks like the frame pointer issue is gone (at least in part) so this can be merged now.

**Comment by pgavlin** [comment]
Done. Thanks for doing this!

**Comment by CarolEidt** [comment]
Yes, thanks!

**Pull request merged**